Services/CO.CDP.Organisation.WebApi/UseCase/UpdateOrganisationUseCase.cs

```diff
@@ -97,24 +97,36 @@ public async Task<bool> Execute((Guid organisationId, UpdateOrganisation updateO
 
                 break;
             case OrganisationUpdateType.AdditionalIdentifiers:
-                if (updateObject.AdditionalIdentifiers == null)
+                if (updateObject.AdditionalIdentifiers == null || !updateObject.AdditionalIdentifiers.Any())
                 {
                     throw new InvalidUpdateOrganisationCommand("Missing additional identifiers.");
                 }
 
                 foreach (var identifier in updateObject.AdditionalIdentifiers)
                 {
+                    if (string.IsNullOrWhiteSpace(identifier.Id))
+                    {
+                        throw new InvalidUpdateOrganisationCommand($"Missing Identifier Number for scheme '{identifier.Scheme}'.");
+                    }
+
+                    // Check if Identifier number already exists
+                    var organisationIdentifier = organisationRepository.FindByIdentifier(identifier.Scheme, identifier.Id);
```

Comment:
Why is this query necessary since we already have the organisation object with identifiers fetched?

Reply:
it is necessary to check, if identifier number already assigned to different organisation.

Reply:
@shilpigoel-goaco and this is new logic not covered by tests